### DISCUSSION TITLE
Linking to labels

### DISCUSSION THREAD

**Discussion by bryceerobertson** [danniehansen/activity-timer#62] [Ideas]
Hi Dannie,

Awesome power-up, thanks a lot! One thing which would be awesome to have: to be able to trigger the timer to start when I add a certain label to a card. At my company we have a card called "Doing Now", and it would be great if the timer would automatically start when we add that card, and even better when it then stops when the card is removed.

I tried doing this with Butler in Trello but unfortunately Activity Timer doesn't show up as an option.

Many thanks in advance!

Bryce Robertson

**Comment by bryceerobertson** [comment]
Hey Dannie, just checking in on whether this would be possible?

Cheers,
Bryce

**Reply by danniehansen** [reply]
Hi @ricewithabee,

Sorry for the delayed response. I've had a rough few weeks.

I've looked into the technical side of this request and unfortunately Trello's powerup tools doesn't allow for me to detect when a specific user changes labels on a card. I can generally make a detection if labels change - but not by whom. This was a similar issue i encountered when making the feature where it automatically starts the timer when moving it into a specific list. It was quite the technical challenge to solve this as i would have to build up infrastructure that would listen in using a webhook to detect changes to a card & by whom.

Similarly one could do the same here for the label. I'm not sure how wide the use case would be for other users & how much usage i would see with such an investment.

For now I'll leave the discussion open for others to pitch in their thoughts. Please if you do want this as a feature, then +1 the post

**Reply by bryceerobertson** [reply]
Hi Dannie,

Thanks so much for such a detailed response, I appreciate it!

I think I may not have been clear enough – for us at my workplace at least, it doesn't matter _who_ changes the label, it's just about the label being changed. So all we would need is for the timer to automatically start when anyone adds a certain label (in our case, the label called "Doing Now") and then to stop automatically when anyone removes that label.

Does that make sense? Hopefully that means it's an easy add-on from the technical side of things.

Thanks again,
Bryce

**Reply by danniehansen** [reply]
Hi @ricewithabee,

Yeah - I got that part. The tricky thing is the timer has to start for the person who adds the label :) Each time tracking is assigned to a user - so we have to know which user added a label to know from whom to start the timer. This is the tricky part unfortunately.

**Reply by bryceerobertson** [reply]
Ah I see the problem now. Yeah that makes sense.

In case it helps, in our case we simply have one list per person, so the system would just need to know what list the card is in in order to see what person had changed the label. But I can imagine that a lot of your customers don't have the same system, so I'm not sure if that is helpful.

Well thank you anyway for your responses, I really appreciate it!

And I hope the last couple weeks for you have been better than those before :)

Cheers,
Bryce
